File: peek_plugin_graphdb/tuples/GraphDbTraceResultTuple.py
from typing import List, Optional

from vortex.Tuple import Tuple, addTupleType, TupleField

from peek_plugin_graphdb._private.PluginNames import graphDbTuplePrefix
from peek_plugin_graphdb.tuples.GraphDbImportEdgeTuple import GraphDbImportEdgeTuple
from peek_plugin_graphdb.tuples.GraphDbImportVertexTuple import GraphDbImportVertexTuple


@addTupleType
class GraphDbTraceResultTuple(Tuple):
    """ GraphDB Trace Result Tuple

    This tuple contains the result of running a trace on the model

    """
    __tupleType__ = graphDbTuplePrefix + 'GraphDbTraceResultTuple'

    #:  The key of the model set that the result was created with.
    modelSetKey: str = TupleField()

    #:  The key of the Trace Config
    traceConfigKey: str = TupleField()

    #:  The key of the vertex start point of this trace
    startVertexKey: str = TupleField()

    #:  The edges
    edges: List[GraphDbEdgeTuple] = TupleField([])

    #:  The vertexes
    vertexes: List[GraphDbImportVertexTuple] = TupleField([])

    #:  The message if the trace was aborted
    traceAbortedMessage: Optional[str] = TupleField()

File: peek_plugin_graphdb/plugin-module/GraphDbTraceResultTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {graphDbTuplePrefix} from "./_private/PluginNames";
import {GraphDbModelSetTuple} from "./GraphDbModelSetTuple";
import {GraphDbEdgeTuple} from "./GraphDbEdgeTuple";
import {GraphDbVertexTuple} from "./GraphDbVertexTuple";
import {GraphDbSegmentLinkTuple} from "./GraphDbSegmentLinkTuple";


@addTupleType
export class GraphDbTraceResultTuple extends Tuple {
    public static readonly tupleName = graphDbTuplePrefix + "GraphDbTraceResultTuple";

    //  The key of the model set that the resuslt was created with.
    modelSetKey: string;

    //  The key of the Trace Config
    traceConfigKey: string;

    //  The key of the vertex start point of this trace
    startVertexKey: string;

    //  The edges
    edges: GraphDbEdgeTuple[];

    //  The vertexes
    vertexes: GraphDbVertexTuple[];

    constructor() {
        super(GraphDbTraceResultTuple.tupleName)
    }
}